ReactSkia/sdk/Alert.cpp

```diff
@@ -0,0 +1,230 @@
+    return;
+  }
+
+  alertInfo alertRef=alertInfoList_.back();
+  idOfMessageOnDisplay_= alertInfoList_.size();
+  double drawStartPointX{0},drawStartPointY{0};
+
+  SkIRect msgAndTitleDirtyRect;
+  pictureCanvas_->clear(DARK_THEME_BACKGROUND_COLOR);
+
+  if (!alertRef.alertTitle.empty()) {
+
+    double titleWidth=font_.measureText( alertRef.alertTitle.c_str(),
+                                         alertRef.alertTitle.length(),
+                                         SkTextEncoding::kUTF8);
+    drawStartPointX=(alertWindowSize_.width() - titleWidth)/2;
+    drawStartPointY=(alertWindowSize_.height() - textFontSize_) / 2;
+
+    pictureCanvas_->drawSimpleText(
+                              alertRef.alertTitle.c_str(),
+                              alertRef.alertTitle.length(),
+                              SkTextEncoding::kUTF8,
+                              drawStartPointX,
+                              drawStartPointY,
+                              font_,
+                              paint_);
+    setWindowTittle(alertRef.alertTitle.c_str());
+
+    msgAndTitleDirtyRect =SkIRect::MakeXYWH(drawStartPointX,
+                                            drawStartPointY-textFontSize_,
+                                            titleWidth,
+                                            textFontSize_+lineSpacing_);
+
+    RNS_LOG_DEBUG(" drawAlert Title :: "<<alertRef.alertTitle.c_str());
+  }
+
+  if (!alertRef.alertMessage.empty()) {
+
+    double messageWidth=font_.measureText( alertRef.alertMessage.c_str(),
+                                         alertRef.alertMessage.length(),
+                                         SkTextEncoding::kUTF8);
+    drawStartPointY += textFontSize_ + lineSpacing_;
+    drawStartPointX=(alertWindowSize_.width() - messageWidth)/2;
+
+    pictureCanvas_->drawSimpleText(
+                            alertRef.alertMessage.c_str(),
+                            alertRef.alertMessage.length(),
+                            SkTextEncoding::kUTF8,
+                            drawStartPointX,
+                            drawStartPointY,
+                            font_,
+                            paint_);
+
+    msgAndTitleDirtyRect.join(SkIRect::MakeXYWH(drawStartPointX,drawStartPointY-textFontSize_,
+                               messageWidth,textFontSize_+lineSpacing_));
+    RNS_LOG_DEBUG(" drawAlert Msg :: "<<alertRef.alertMessage.c_str());
+  }
+
+  dirtyRect.push_back(msgAndTitleDirtyRect);
+}
+
+inline void Alert::removeAlertFromAlertList(unsigned int msgIndex) {
+
+  if(alertInfoList_.empty() || (msgIndex > alertInfoList_.size())) {
+    return;
+  }
+
+  if(alertInfoList_.size() == idOfMessageOnDisplay_) {
+    // Current Alert Message is the Latest Message
+    alertInfoList_.pop_back();
+  } else {
+    //New Alert Message Pending to be displayed
+    std::list<alertInfo>::iterator it = alertInfoList_.begin();
+    std::advance(it, msgIndex);
+    alertInfoList_.erase(it);
```

Comment:
Binary addition is possible in case of vector iterator, NOt possible for list. SO advance function or unary increment are only possible for List.   
Both List & Vector uses different iterators